### PR TITLE
GPC-364 Add event generation to fake gro

### PR DESCRIPTION
**Pull request by CarlyG55**

### CODE REVIEW

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                PersonNameSuffix: "",
+            },
+            DeceasedAliasNameType: "",
+            DeceasedMaidenNameType: "",
+            DeceasedGender: getRandomElement(genderOptions),
+            DeceasedDeathDate: {
+                PersonDeathDate: getRandomElement(dateOfDeathOptions),
+                VerificationLevel: "03",
+            },
+            PartialMonthOfDeath: "",
```
Comment (olevett) [inline terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts:100]: Should we be populating these fields from the DeathDate?

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    ["10 Test Street", "Test Town", "Test County"],
+    ["", "Westminster", "London"],
+    ["27 Brown Lane", "", "Somerset"],
+    ["65 Link Road", "Southport", ""],
+]
+const postcodeOptions = ["PR8 1HY", ""]
+
+const handler: Handler = async (event): Promise<InsertXmlResponse> => {
+    const numberOfRecords =
+        event["detail-type"] === "Scheduled Event" || !event["numberOfRecords"] ? 25 : event["numberOfRecords"]
```
[the remark this answers is not in the document]
Comment (CarlyG55) [reply terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts:29]: I'm pretty sure it's a default - I checked through the len scheduled lambda and couldn't find anything, and this is what that uses

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts (hunk 1 above)
Comment (CarlyG55) [reply terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts:100]: I filled in the sections that were filled in in the example schema - [here](https://docs.google.com/spreadsheets/d/1fFMlFL52P2FYAd5X-0PlGbKuPn-4uEkn/edit#gid=1526528896) someone seemed to ask if these should be calculated, but that seems to be on the DWP end?

Comment (oskarwilliams) [reply terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts:100]: So at the moment we are passing this through so I would suggest adding it, once we get the sample JSON back from DWP we will have to update all our models so thats ok to update then

File: terraform/demo/main.tf [hunk 3]
```diff
@@ -116,4 +116,6 @@ module "gro_ingestion_service" {
   auth_url                = module.data_share_service.token_auth_url
   publisher_client_id     = module.data_share_service.gro_ingestion_client_id
   publisher_client_secret = module.data_share_service.gro_ingestion_client_secret
+
+  insert_xml_lambda_schedule = "rate(5 minutes)"
```
Comment (oskarwilliams) [inline terraform/demo/main.tf:120]: could we do 1 hours for the moment in both envs, we can update to 5 minutes when these get actively used

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    const putObjectCommand = new PutObjectCommand({
+        Bucket: config.s3BucketArn,
+        Key: fileKey,
+        Body: xml,
+    })
+    await s3Client.send(putObjectCommand)
+}
+
+const generateXml = (numberOfRecords: number) => {
+    const deathRecords = [...Array(numberOfRecords)].map(_ => createDeathRecord())
```
Comment (oskarwilliams) [inline terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts:60]:
```suggestion
    const deathRecords = [...Array(numberOfRecords)].map(createDeathRecord)
```

I _think_ you don't need the `_ => `

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**